addons/mass_mailing/models/mass_mailing.py

```diff
@@ -341,15 +356,7 @@ def get_recipients(self, cr, uid, ids, model=None, context=None):
         return res
 
     def on_change_campaign_name(self, cr, uid, ids, name, context=None):
-        if name:
-            mass_mailing_campaign = self.browse(cr, uid, ids, context=context)
-            if mass_mailing_campaign.campaign_id:
-                utm_campaign_id = mass_mailing_campaign.campaign_id.id
-                self.pool['utm.campaign'].write(cr, uid, [utm_campaign_id], {'name': name}, context=context)
-            else:
-                utm_campaign_id = self.pool['utm.campaign'].create(cr, uid, {'name': name}, context=context)
-
-            return {'value': {'campaign_id': utm_campaign_id}}
+        return {'value': {}}
```

Comment:
Just to tourner en rond: if new code and not -u, I think it will crash. Your old code about searching a random campaign was probably necessary.